### PR TITLE
drraw2trace. Change the type of error codes from string to int.

### PR DESCRIPTION
**Pull request by yuriershov**
Error codes type changed from string to int. Performance of drraw2trace is increased ~15%.

### CODE REVIEW

File: clients/drcachesim/tracer/raw2trace.h [hunk 1]
```diff
@@ -1188,16 +1268,16 @@ class raw2trace_t {
      * all buffers from any one traced thread must be processed by the same worker
      * thread, both for correct ordering and correct synchronization.
      */
-    std::string
+    raw2trace_error_t
```
[the remark this answers is not in the document]
Comment (derekbruening) [reply clients/drcachesim/tracer/raw2trace.h:1271]: What do you think of a solution where `bool` is the return value with `false` indicating an error and a separate `get_error_string()` which returns a descriptive string?  That then allows our subclass (and anyone else's) to have new error types without having to try to add them to the public base class.  It also eliminates the separate string array which has maintenance costs for keeping it in sync.  It should have the same performance of the enum solution.

File: clients/drcachesim/tracer/raw2trace.cpp [hunk 2]
```diff
@@ -2437,11 +2391,9 @@ raw2trace_t::adjust_and_emit_rseq_buffer(raw2trace_thread_data_t *tdata, addr_t
     tdata->rseq_buffering_enabled_ = false;
 
     log(4, "Writing out rseq buffer: %zd entries\n", tdata->rseq_buffer_.size());
-    auto success =
-        write(tdata, &tdata->rseq_buffer_[0], &tdata->rseq_buffer_.back() + 1,
-              tdata->rseq_decode_pcs_.data(), tdata->rseq_decode_pcs_.size());
-    if (!success)
-        return success;
+    if (! write(tdata, &tdata->rseq_buffer_[0], &tdata->rseq_buffer_.back() + 1,
```
Comment (derekbruening) [inline clients/drcachesim/tracer/raw2trace.cpp:2394]: nit: Extra space after `!` (surprised clang-format doesn't complain)

File: clients/drcachesim/tracer/raw2trace.cpp [hunk 3]
```diff
@@ -566,7 +566,6 @@ raw2trace_t::process_offline_entry(raw2trace_thread_data_t *tdata,
                                    OUT bool *end_of_record, OUT bool *last_bb_handled,
                                    OUT bool *flush_decode_cache)
```
Comment (derekbruening) [inline clients/drcachesim/tracer/raw2trace.cpp:567]: I think some tests are failing b/c of too-long >90 lines -- please fix to get the tests green.

**Review by derekbruening — APPROVED** [review]
LGTM with just a couple of nits.  Thank you for making the changes.

**Comment by kuhanov** [comment]
Hi, @derekbruening. Could you add Yury (@yuriershov) to contributor list? He's going to share a few patches.
Thx, Kirill

**Comment by kuhanov** [comment]
> The title and description are now inaccurate: they say "int" when it's actually "bool". Please remember to update when the code changes.

Oh, ok.

**Pull request merged**